### PR TITLE
New option: logging-format-style for logging checker

### PR DESCRIPTION
**Pull request by pinealan**
**Steps**
- [x] Add myself to contributors
- [x] Add to changelog
- [x] Add to docs/whatsnew/2.2.rst

**Description**
`logging-format-style` accepts one of '%' or '{', (defaults to '%'). When '{' is selected, logging checker assumes str.format() style format strings for calls to the logging.

This PR came about when pylint threw E1205 `logging-too-many-args` at my project that uses '{' style format strings. This was because pylint was unable to count the required number of args for the format string. The new feature indirectly fixes that by allowing the proper interpretation of that format string. The possibility for others to implement the '$' style template strings parsing is also open.

### CODE REVIEW

File: pylint/checkers/utils.py [hunk 1]
```diff
@@ -535,6 +535,67 @@ def next_char(i):
     return keys, num_args, key_types, pos_types
 
 
+def parse_brace_format_string(format_string: str) -> Tuple[Set[str], int]:
```
Comment (PCManticore) [inline pylint/checkers/utils.py]: It's not totally obvious but we have `parse_format_method_string` in `checkers/strings.py`, which does the same thing. It's used by the string checkers for warning about passing invalid format keys when formatting with `.format()`.

Comment (pinealan) [reply pylint/checkers/utils.py]: Thanks for pointing that out. I will refactor accordingly.

File: pylint/checkers/utils.py [hunk 2]
```diff
@@ -535,6 +536,104 @@ def next_char(i):
     return keys, num_args, key_types, pos_types
 
 
+if PY3K:
```
Comment (PCManticore) [inline pylint/checkers/utils.py]: Since we're now running only on Python 3, feel free to remove the `else` clause of this check.

**Comment by PCManticore** [comment]
Thanks a lot @pinealan !

**Pull request merged**